src/Context/AuthContext.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, useEffect, useState } from "react";
+import { jwtDecode } from "jwt-decode";
+
+
+
+export  const AuthContext = createContext ({});
+
+
+export default function AuthContextProvider(props:any){
+
+  const [userData, setUserData] = useState(null);
+  const[role,setRole]=useState(null)
+  
+
+  let reqHeaders={
+    Authorization:`Bearer ${localStorage.getItem("userToken")}`
+  }
+  const baseUrl = "http://upskilling-egypt.com:3000/api/v0";
+
+  const saveUserData = () => {
+    const encodedToken:any = localStorage.getItem("userToken");
+    const decodedToken:any = jwtDecode(encodedToken)
```

Comment:
We don't need to use jwtDecode, as the data coming from the backend when we login is what we get from the decoded token.